### PR TITLE
[RFC] provider/pythonx: handle exit code 127 from pyenv

### PR DESCRIPTION
**Pull request by blueyed**
This also checks the major/min version only for expected return codes.

With pyenv, you might get the following (return code 127):

```
pyenv: python3.4: command not found

The `python3.4' command exists in these Python versions:
  3.4.3
  3.4.3/envs/tmp-3.4.3-eElS6Y
  tmp-3.4.3-eElS6Y
```

/cc @shougo @fwalch

### CODE REVIEW

File: runtime/autoload/provider/pythonx.vim [hunk 1]
```diff
@@ -83,8 +83,8 @@ function! s:check_interpreter(prog, major_ver, skip) abort
   " Try to load neovim module, and output Python version.
   " Return codes:
   "   0  Neovim module can be loaded.
-  "   1  Something else went wrong.
```
Comment (justinmk) [inline runtime/autoload/provider/pythonx.vim]: we still return 1 on line 119

Comment (blueyed) [inline runtime/autoload/provider/pythonx.vim]: This comment refers to the system call, doesn't it?

Comment (justinmk) [inline runtime/autoload/provider/pythonx.vim]: Not sure. I thought it was documenting `s: s:check_interpreter`, but the placement here would be odd.

**Comment by justinmk** [comment]
LGTM

**Pull request merged**